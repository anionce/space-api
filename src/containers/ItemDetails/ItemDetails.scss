@import '../../assets/styles/variables';

.container {
	margin-top: 20px;
	display: flex;
	flex-direction: column;
	align-items: center;
	text-align: center;

	.title {
		font-size: $font-size-lg;
		font-weight: 500;
		max-width: 250px;
	}

	.subcontainer {
		box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
		display: flex;
		width: 500px;
		flex-direction: column;
		align-items: center;
		background-color: $color-white;
		padding: 20px;
		border: 0px solid $color-primary;
		border-radius: 10px;

		@media (min-width: $sm) {
			max-width: 260px;
		}
	}
}
