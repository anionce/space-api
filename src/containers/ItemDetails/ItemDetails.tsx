import { Link, useLocation, useParams } from 'react-router-dom';
import { useAppSelector } from '../../store/hooks';
import './ItemDetails.scss';
import { useGetCapsuleByIdQuery } from '../../services/api';
import { Rocket } from '../../models/Rocket';
import React from 'react';

export const ItemDetails = () => {
	const { id } = useParams();
	const { pathname } = useLocation();

	const isCapsule: Boolean = pathname.includes('capsules');

	const getAllRockets: Rocket[] = useAppSelector(state => state.rocketSlice.rockets);
	const rocket: Rocket | undefined = getAllRockets.find(rocket => rocket.id === id);
	const { data: capsule, isFetching, isError } = useGetCapsuleByIdQuery(id);

	return (
		<>
			{capsule && isFetching && <div>Loading...</div>}
			{capsule && isError && <div>There has been an error loading the capsule</div>}
			<div className='container'>
				<div className='subcontainer'>
					<div className='title'>{isCapsule ? capsule?.type : rocket?.type}</div>
					<p>{isCapsule ? capsule?.type : rocket?.type}</p>
					{!isCapsule && (
						<Link to={`/rockets/${id}/update/`}>
							<button>Edit</button>
						</Link>
					)}
				</div>
			</div>
			)
		</>
	);
};
