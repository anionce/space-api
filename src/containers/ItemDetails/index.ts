import { ItemDetails } from './ItemDetails';
export { ItemDetails };
