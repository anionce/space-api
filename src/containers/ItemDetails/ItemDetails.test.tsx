import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { ItemDetails } from './ItemDetails';

test('renders correcly', () => {
	const { container } = customRender(<ItemDetails />);

	expect(container).toMatchSnapshot();
});

test('all elements are displayed', () => {
	customRender(<ItemDetails />);

	expect(screen.getByText(/Edit/i)).toBeInTheDocument();
});
