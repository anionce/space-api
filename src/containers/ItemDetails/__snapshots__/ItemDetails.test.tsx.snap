// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="subcontainer"
    >
      <div
        class="title"
      />
      <p />
      <a
        href="/rockets/undefined/update/"
      >
        <button>
          Edit
        </button>
      </a>
    </div>
  </div>
  )
</div>
`;
